tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_delete_thread.py

```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import SkipTest
+from marionette_driver import Wait
+from gaiatest import GaiaTestCase
+from gaiatest.apps.messages.app import Messages
+from gaiatest.apps.system.app import System
+from gaiatest.utils.plivo.plivo_util import PlivoUtil
+
+
+class TestSms(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+
+        try:
+            self.testvars['plivo']
+        except KeyError:
+            raise SkipTest('Plivo account details not present in test variables')
+
+        plivo = PlivoUtil(
+            self.testvars['plivo']['auth_id'],
+            self.testvars['plivo']['auth_token'],
+            self.testvars['plivo']['phone_number']
+        )
+
+        plivo.send_sms(to_number=self.environment.phone_numbers[0].replace('+', ''),
```

Comment:
I don't see why it's wrong to be dependant from an internal API here.
We use these data_layer things all the time to get into a state from where we want to start test from. We could also do all the manual steps in automation to send the SMS, but that would make the test much slower, to get to the part which we want to test here.
The only thing that we want to test here is the delete thread feature in SMS. We could also create the entries in the SMS database directly, for instance, to work from there to test the delete thread feature.